checkov/terraform/checks/resource/aws/IAMUserNotUsedForAccess.py

```diff
@@ -0,0 +1,30 @@
+from typing import List, Any
+
+from checkov.common.models.consts import ANY_VALUE
+from checkov.common.models.enums import CheckCategories
+from checkov.terraform.checks.resource.base_resource_negative_value_check import BaseResourceNegativeValueCheck
+
+
+class IAMUserNotUsedForAccess(BaseResourceNegativeValueCheck):
+    def __init__(self) -> None:
+        name = "Ensure IAM users are not defined " \
+               "AWS Access should be controlled by roles via an SSO, inline with AWS security best practices " \
+               "Require human users to use federation with an identity provider to access AWS using temporary credentials "
+        id = "CKV_AWS_272"
+        supported_resources = ('aws_iam_user', 'aws_iam_policy_attachment')
```

Comment:
reading the description it doesn't make sense to also check `aws_iam_policy_attachment` because this resource doesn't add a user it is "just" giving permissions to it. This could be a different check, which doesn't make any sense, like `Ensure IAM users have no permissions` 😄

Reply:
updated with the new commit 107bd3a